compat/src/main/scala-2.11_2.12/scala/collection/compat/PackageShared.scala

```diff
@@ -55,7 +55,12 @@ private[compat] trait PackageShared {
 
   implicit def genericCompanionToCBF[A, CC[X] <: GenTraversable[X]](
       fact: GenericCompanion[CC]): CanBuildFrom[Any, A, CC[A]] = {
-    val builder: m.Builder[A, CC[A]] = fact match {
+    /* see https://github.com/scala/scala-library-compat/issues/337
+       `simpleCBF.apply` takes a by-name parameter and relies on
+       repeated references generating new builders, thus this expression
+       must be non-strict
+     */
+    def builder: m.Builder[A, CC[A]] = fact match {
```

Comment:
could probably optimise this so that it doesn't have to pattern match each time, and only does so once